test/common.py

```diff
@@ -152,6 +157,7 @@ def is_iterable(obj):
 class TestCase(unittest.TestCase):
     precision = 1e-5
     maxDiff = None
+    doCUDAMemoryCheck = True
```

Comment:
Why is this set to `True` by default? Shouldn't it be only enabled in those test cases that actually deal with CUDA? Otherwise it's going to slow down a whole bunch of other things as well, and it might be entirely unnecessary. Our test suite is already unbearably slow, let's not make it 3x worse.